### DISCUSSION TITLE
Package request - OpenCV

### DISCUSSION THREAD

**Discussion by RSchueder** [microsoft/PlanetaryComputer#44] [Q&A] [answered]
I was hoping to have OpenCV added to the MPC image. Installation of this package requires the ability to install system libraries and this cannot be done through the terminal available to users in MPC.

In my own pangeo environments I have installed it using the following steps:

In Dockerfile:
```
RUN apt-get update && apt-get install -y
RUN apt-get install python3-opencv -y
```

in environment.yml, under pip dependancies
```
 - pip:
   - opencv-python==4.5.5.62 
```

This would help us folks trying to use classic computer vision algorithms on satellite images. If others know of alternative implementations of morphological dilation filters already available through packages presently on MPC, please let me know.

**Comment by TomAugspurger** [comment, marked as answer]
Thanks for opening this. Does https://scikit-image.org/docs/stable/api/skimage.morphology.html#skimage.morphology.dilation, via scikit-image, suite your needs? As an alternative, to the Dockerfiles, you might be able to use `mamba install -c conda-forge opencv`.

I can look into adding it to the base image. The tradeoff is the increased size of the Dockerfile and potential conflicts with other libraries present in the image.

**Reply by RSchueder** [reply]
Thanks for the reply Tom. I will give it a try and feed back. Cheers.

**Reply by RSchueder** [reply]
Turns out with just a little bit of reading the docs I can reproduce the workflow using skimage, so please consider this issue closed for my particular ask.

**Reply by TomAugspurger** [reply]
Thanks for confirming! If there are other operations only implemented in opencv, then let me know and we can re-consider adding it.

**Reply by 007anupamdas** [reply]
There are some specific functions which are exclusive to opencv. I tried using scikit, but it didn't meet my demands. Can you please add it in base release. I found new libraries like rioxarray which is not even renowned like opencv, in image processing domain, and not of much use, as I use rasterio for much of my raster processing.
